app/components/filter-plaques/component.js

```diff
@@ -26,18 +26,20 @@ export default Ember.Component.extend({
       // Fetch display names
       if (ids.length > 0) {
         this.fetchAgentDetails(ids).then((data) => {
-            var displayFilters = filters.map((filter) => {
-              var value = filter.value;
-              data.forEach((agentData) => {
-                if (value === agentData.id) {
-                  value = agentData.name;
-                }
-              });
-              return {key: filter.key, value: value};
-            });
-            this.set('filters', displayFilters);
+          var displayFilters = filters.map((filter) => {
+            var value = filter.value;
+            for (let agentData of data) {
+              if (value === agentData.id) {
```

Comment:
For each filter, (e.g. `{key: 'contributor', value: 3}`, we find the equivalent record in `data` (e.g. `{id: 3, name: 2}`) and replace the value with the `name`. It should not matter that the name is the value of another tag since the record is compared by id.

Here's some sample code I used to double check: 

```
filters = [{key: 'contributor', value: 3}, {key: 'publisher', value: 2}, {key: 'funder', value: 1}];
data = [{id: 1, name: 'Mary'}, {id: 2, name: 'Joe'}, {id: 3, name: 2}]
var displayFilters = filters.map((filter) => {
    var value = filter.value;
        for (let agentData of data) {
            if (value === agentData.id) {
                value = agentData.name;
                break;
              }
            }
            return {key: filter.key, value: value};
});
displayFilters = [{key: 'contributor', value: 2}, {key: 'publisher', value: 'Joe'}, {key: 'funder', value: 'Mary'}];
```